### PR TITLE
fix(service): control parentId when creating page

### PR DESCRIPTION
**Pull request by jourdiw**
## Issue

https://gravitee.atlassian.net/browse/APIM-3049

## Description

- Transform empty parentId to null for page creation
- Check if parent is folder for page creation

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/usecase/ApiCreateDocumentationPageUsecase.java [hunk 1]
```diff
@@ -69,12 +71,18 @@ public record Output(Page createdPage) {}
...
         var parentId = page.getParentId();
+
         if (Objects.nonNull(parentId) && !parentId.isEmpty()) {
-            try {
-                pageCrudService.get(parentId);
-            } catch (PageNotFoundException ignored) {
-                page.setParentId(null);
+            var foundParent = pageCrudService.findById(parentId);
+
+            if (foundParent.isPresent()) {
```
Comment (loriepisicchio) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/usecase/ApiCreateDocumentationPageUsecase.java:78]: Suggestion: also throw exception if parent does not exist

Comment (jourdiw) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/usecase/ApiCreateDocumentationPageUsecase.java:78]: In the old service, if the parent didn't exist, we just didn't save the parent id. I'm tempted to leave it like this for now.

Comment (loriepisicchio) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/usecase/ApiCreateDocumentationPageUsecase.java:78]: Then maybe a test to illustrate this behavior ? I'm afraid this is not explicit enough and we will wonder why later.

Comment (jourdiw) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/usecase/ApiCreateDocumentationPageUsecase.java:78]: Would this test work for you? https://github.com/gravitee-io/gravitee-api-management/pull/5642#discussion_r1368667278

Comment (jourdiw) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/usecase/ApiCreateDocumentationPageUsecase.java:78]: Oops. Here's the right test: https://github.com/gravitee-io/gravitee-api-management/pull/5642#discussion_r1368667278

**Review by loriepisicchio — APPROVED** [review]

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/usecase/ApiCreateDocumentationPageUsecase.java (hunk 1 above)
Comment (jourdiw) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/usecase/ApiCreateDocumentationPageUsecase.java:78]: Link to test: https://github.com/gravitee-io/gravitee-api-management/blob/b2320637529bd022e67e2f46988d14edccf3ecbe/gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/documentation/usecase/ApiCreateDocumentationPageUsecaseTest.java#L243

**Pull request merged**